core/arch/arm/include/kernel/user_access_arch.h

```diff
@@ -23,4 +23,26 @@ static inline void enter_user_access(void) {}
 static inline void exit_user_access(void) {}
 #endif /* CFG_PAN */
 
+#ifdef ARM64
+#define GET_USER(_x, _p) ({						\
+	TEE_Result __res = TEE_SUCCESS;					\
+	__typeof(_x) __x = 0;						\
+	static_assert(sizeof(_x) == sizeof(*(_p)));			\
+	__res = copy_from_user(&__x, (const void *)_p, sizeof(__x));	\
+	(_x) = (__x);							\
+	__res;								\
+})
+
+#define PUT_USER(_x, _p) ({						\
+	TEE_Result __res = TEE_SUCCESS;					\
+	__typeof(*(_p)) __x = (_x);					\
+	static_assert(sizeof(_x) == sizeof(*(_p)));			\
+	__res = copy_to_user((void *)_p, &(__x), sizeof(*(_p)));	\
+	__res;								\
+})
+#else
+#define GET_USER(_x, _p) ({ (_x) = *(_p); TEE_SUCCESS; })
+#define PUT_USER(_x, _p) ({ *(_p) = (_x); TEE_SUCCESS; })
```

Comment:
These ARM32 macros definitly need to use `copy_from_user()`/`copy_to_user()` as later change remove access right check as expected done from `GET_USER()`/`PUT_USER()`.
Ditto for the RISCV versions of the macros.